src/TTextEdit.cpp

```diff
@@ -993,13 +994,49 @@ void TTextEdit::mouseMoveEvent( QMouseEvent * event )
         }
     }
     x = ( event->x() / mFontWidth ) - timeOffset;
-    if( ( x < 0 ) || ( y < 0 ) || ( y > (int) mpBuffer->size()-1 ) )
+    if( ( y < 0 ) || ( y > (int) mpBuffer->size()-1 ) )
```

Comment:
I actually read about that sort of thing when going through some of Qt's documentation once. The thing here is that most of these uses of `[ ]` are on things that we are certain already exist. For example, all the calls to `buffer[y][x]` are placed after conditionals that prevent it from being accessed if `buffer[y][x]` does not exist yet. I'm definitely trying to keep `.at( )` in mind as often as I can though.

Now, as far as the error, it could be related to `x` being less that 0 I guess? I had the mouseMoveEvent function full of debug messages that would print to the output so I could hopefully see where it was dying, but it seemed like it was crashing either before or after the function runs in its entirety, cause I saw my **mouseMoveEvent>** echo and my **mouseMoveEvent end>** echo both. Unlike the timestamp thing, this bug is NOT something I can replicate in a blank dev branch, but it's only happening when I'm NOT using ctrlSelecting. That's what's confusing me, since the code running should be the same if `mCtrlSelecting` isn't true. 😕